src/main/java/cr0s/warpdrive/config/structures/Schematic.java

```diff
@@ -65,28 +102,142 @@ public boolean loadFromXmlElement(final Element element) throws InvalidXmlExcept
 			} catch (final InvalidXmlException exception) {
 				exception.printStackTrace(WarpDrive.printStreamError);
 				WarpDrive.logger.error(String.format("Skipping invalid insertion %s",
-				                                     blockState));
+				blockState));
 			}
 		}
 		
 		return true;
 	}
 	
-	@Override
-	public boolean generate(@Nonnull final World world, @Nonnull final Random random, @Nonnull final BlockPos blockPos) {
-		return instantiate(random).generate(world, random, blockPos);
-	}
+	static class BlockMatcher {
+		
+		Block block;
+		IBlockState blockState;
+		boolean blockStateWildcard;
+		
+		public static BlockMatcher fromBlockStateString(String blockStateString) {
+			///TODO: allow different data input type for meta: range (e.g. 2-13), comma sepearated list (e.g. 1,2,3..), multiple property (e.g. variant=oak,half=bottom)
+			
+			BlockMatcher result = new BlockMatcher();
+			
+			String state = "";
+			String metaString = "*";
+			if (blockStateString.contains("@")) {    // with meta
+				String[] blockStateParts = blockStateString.split("@");
+				state = blockStateParts[0].trim();
+				metaString = blockStateParts[1].trim();
+			} else {    // without meta
+				state = blockStateString;
+			}
+			result.block = Block.getBlockFromName(state);
+			
+			if (result.block == null) {
+				WarpDrive.logger.warn(String.format("Found invalid block in replacement or insertion: %s", state));
```

Comment:
Consider passing a context string instead of assuming caller is replacement or insertion? also, this allows to provide more details on where the faulty data comes from.